dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -179,23 +179,93 @@ def _add_helper_merge_profile_histograms(
+        global_min_of_histogram_edges = (
+            float(self.min)
+            if self.min is not None
+            else min(
+                other1._stored_histogram["histogram"]["bin_edges"][0],
+                other2._stored_histogram["histogram"]["bin_edges"][0],
+            )
         )
-        bin_counts, bin_edges = self._get_histogram(combined_values)
-        self._stored_histogram["histogram"]["bin_counts"] = bin_counts
-        self._stored_histogram["histogram"]["bin_edges"] = bin_edges
 
-        histogram_loss = self._histogram_bin_error(combined_values)
-        self._stored_histogram["histogram"]["current_loss"] = histogram_loss
-        self._stored_histogram["histogram"]["total_loss"] = histogram_loss
+        global_max_of_histogram_edges = (
+            float(self.max)
+            if self.max is not None
+            else max(
+                other1._stored_histogram["histogram"]["bin_edges"][-1],
+                other2._stored_histogram["histogram"]["bin_edges"][-1],
+            )
+        )
+
+        # Generate new bin edges
+        ideal_bin_edges = np.linspace(
+            global_min_of_histogram_edges,
+            global_max_of_histogram_edges,
+            ideal_count_of_bins + 1,
+        )
+
+        # Initialize new count by bin object for population
+        new_entity_count_by_bin = np.zeros((ideal_count_of_bins,))
+
+        # Generate new histograms
+        _, histogram_loss_1 = self._assimilate_histogram(
+            from_hist_entity_count_per_bin=other1._stored_histogram["histogram"][
+                "bin_counts"
+            ],
+            from_hist_bin_edges=other1._stored_histogram["histogram"]["bin_edges"],
+            dest_hist_entity_count_per_bin=new_entity_count_by_bin,
+            dest_hist_bin_edges=ideal_bin_edges,
+            dest_hist_num_bin=ideal_count_of_bins,
+        )
+
+        # Ensure loss is calculated on second run of regenerate
+        _, histogram_loss_2 = self._assimilate_histogram(
+            from_hist_entity_count_per_bin=other2._stored_histogram["histogram"][
+                "bin_counts"
+            ],
+            from_hist_bin_edges=other2._stored_histogram["histogram"]["bin_edges"],
+            dest_hist_entity_count_per_bin=new_entity_count_by_bin,
+            dest_hist_bin_edges=ideal_bin_edges,
+            dest_hist_num_bin=ideal_count_of_bins,
+        )
+        try:
+            calculated_loss = (
+                (histogram_loss_1 * other1.sample_size)
+                + (histogram_loss_2 * other2.sample_size)
+            ) / (other1.sample_size + other2.sample_size)
+        except AttributeError:
+            sample_size_1 = sum(other1._stored_histogram["histogram"]["bin_counts"])
+            sample_size_2 = sum(other2._stored_histogram["histogram"]["bin_counts"])
+            calculated_loss = (
+                (histogram_loss_1 * sample_size_1) + (histogram_loss_2 * sample_size_2)
+            ) / (sample_size_1 + sample_size_2)
```

Comment:
Is `sample_size` the correct value or should it be `match_count`? Do we have tests evaluating this?

Reply:
Also, I think loss should be the sum, right? It's the aggregate loss of both being inserted into the `_stored_histogrtam`.

Reply:
sure match_count is probably right.
so in this calculation we are taking the the loss of each histogram and using the size of the dataset associated with that loss as a weight factor. So yes it is a sum, in this case I am doing a weighted sum.